addons/purchase/models/purchase_order.py

```diff
@@ -564,8 +564,8 @@ def _add_supplier_to_product(self):
             if line.product_id and not already_seller and len(line.product_id.seller_ids) <= 10:
                 price = line.price_unit
                 # Compute the price for the template's UoM, because the supplier's UoM is related to that UoM.
-                if line.product_id.product_tmpl_id.uom_po_id != line.product_uom_id:
-                    default_uom = line.product_id.product_tmpl_id.uom_po_id
+                if line.product_id.product_tmpl_id.uom_id != line.product_uom_id:
+                    default_uom = line.product_id.product_tmpl_id.uom_id
```

Comment:
To check but I'm not sure it's a good practice to modify to product UoM, we could register the seller with the UoM of the line